snap/snapcraft.yaml

```diff
@@ -90,60 +92,85 @@ parts:
         - libxml2
         - libxslt1.1
         - libpython3-stdlib
-        - libpython3.8-stdlib
-        - libpython3.8-minimal
+        - libpython3.10-stdlib
+        - libpython3.10-minimal
         - python3-pip
         - python3-setuptools
         - python3-wheel
         - python3-venv
         - python3-minimal
         - python3-distutils
         - python3-pkg-resources
-        - python3.8-minimal
+        - python3.10-minimal
         - squashfs-tools
         - xdelta3
+    build-attributes:
+      - enable-patchelf
     override-build: |
-      snapcraftctl build
+      ${SNAP}/libexec/snapcraft/craftctl default
 
       echo "Create libsodium symlink..."
       TRIPLET_PATH="$SNAPCRAFT_PART_INSTALL/usr/lib/$(gcc -print-multiarch)"
       LIBSODIUM="$(readlink -n "$TRIPLET_PATH/libsodium.so.23")"
       # Remove so the link can be recreated on re-builds
       rm -f "$TRIPLET_PATH/libsodium.so"
       ln -s "$LIBSODIUM" "$TRIPLET_PATH/libsodium.so"
-      $SNAPCRAFT_PROJECT_DIR/tools/snapcraft-override-build.sh
+
+      # Restore patched files
+      PYTHON_PACKAGE_PATH="${SNAPCRAFT_PART_INSTALL}/usr/lib/python3.10/"
+      CTYPES_INIT="ctypes/__init__.py"
+      CTYPES_INIT_ORIG="patched/${CTYPES_INIT}.orig"
+      [ -f "${CTYPES_INIT_ORIG}" ] && mv "${CTYPES_INIT_ORIG}" "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}"
+
+      # Apply patches
+      echo "Patching ctypes..."
+      patch -s -b "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}" "${SNAPCRAFT_PROJECT_DIR}/patches/ctypes_init.diff"
+
+      # Save patches to allow rebuilding
+      mkdir -p patched/ctypes
+      if [ -f "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}.orig" ]; then
+        mv "${PYTHON_PACKAGE_PATH}/${CTYPES_INIT}.orig" patched/ctypes
+      fi
+
+      # Disable site packages
+      sed -i "${SNAPCRAFT_PART_INSTALL}/usr/lib/python3.10/site.py" \
+        -e 's/^ENABLE_USER_SITE = None$/ENABLE_USER_SITE = False/'
 
   snapcraft:
     source: .
     plugin: python
     python-packages:
         - wheel
         - pip
-        - setuptools<66  # Limited to < 66 because we need `pkg_resources` and because `python-apt` doesn't build with the latest.
-    requirements:
+        # Limited to < 66 because we need `pkg_resources` and because `python-apt`
+        # does not build with the latest.
+        - setuptools<66
```

Comment:
Is this still true?

Reply:
Not sure on the `pkg_resources` front, but `python-apt' 2.4, which comes in core22, seems to work with newer setuptools

Reply:
I need to update this still, I would have but I am focused on figuring out why the configure hook is not working when built here and does when built locally

Reply:
well, we still need `pkg_resources`